area_detector_handlers/handlers.py

```diff
@@ -130,6 +130,18 @@ def close(self):
         self._file = None
 
 
+class HDF5DatasetSliceHandler(HDF5DatasetSliceHandlerPureNumpy):
+    return_type = {'delayed': True}
+
+    def __call__(self, point_number):
+        # Don't read out the dataset until it is requested for the first time.
+        if not self._dataset:
+            self._dataset = self._file[self._key]
```

Comment:
Maybe we want to cache the dask array here rather than generating a new one every time?